core/src/main/kotlin/endpoints/AgentManager.kt

```diff
@@ -220,6 +220,7 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         }.getOrDefault(id, persistentHashMapOf())
         if (instances.isEmpty()) {
             logger.info { "Agent with id '${id}' was disconnected" }
+            entryOrNull(id)?.updateAgent { it.copy(status = AgentStatus.OFFLINE) }?.persistToDatabase()
```

Comment:
why here do we save status in db?

Reply:
Before the changes, we saved all agent statuses changes in the database, but now, we will store only 2 states in the database, either the agent is not registered, or offline , if you do not save the status to offline, then after restarting the system, the agent will rise with the status not registered. 
Although, maybe it's better to save the agent's status offline immediately after registering the agent ?